### ISSUE TITLE
TS2304: Cannot find name 'module'.

### ISSUE THREAD

**Issue by ericmccarthy7** [angular/angular-cli#680]
Info about system & node version
```
OpenSUSE Leap 42.1
angular-cli: 1.0.0-beta.1
node: 5.11.0
os: linux x64
```
Steps taken to produce error
```
ng new test-app
cd test-app
gvim
ng serve
```

Error is via typescript when I'm editing the file app/test-app.component.ts

I can get rid of the error by including
`/// <reference path="../typings.d.ts" />`
at the top of the file.

![image](https://cloud.githubusercontent.com/assets/1617878/15095933/2ced29d8-14b0-11e6-91cc-5d027b8f92d6.png)

![image](https://cloud.githubusercontent.com/assets/1617878/15095935/4c5ff1ba-14b0-11e6-9c53-6f308864d4f2.png)

Using vim with https://github.com/leafgarland/typescript-vim and https://github.com/Valloric/YouCompleteMe

**Comment by ericmccarthy7** [comment]
I can confirm this also is happening on windows in vim -- seems like a specific error to my setup.

**Comment by Namek** [comment]
I've seen live coding on last ng-conf where VSCode couldn't recognize the `module` variable too.

**Comment by Brocco** [comment]
`module` is defined you can see that [here](https://github.com/angular/angular-cli/blob/f1eba5d586cb20a6ea8f62109a0684ce91a7ca5c/addon/ng2/blueprints/ng2/files/__path__/typings.d.ts#L3)

This is probably related to issue #684 you can track the progress there, closing this issue now in favor of the other.

**Comment by iuliust** [comment]
try the following tsconfig filesGlob property : 
`"filesGlob": [
    "main.ts",
    "typings.d.ts",
    "app/**/*.ts",
    "!app/**/*.spec.ts"
  ],`

**Comment by zetalei** [comment]
for atom-typescript :
```json
"files": [
  "main.ts",
  "typings.d.ts"
],
```
for ng build :
```json
"filesGlob": [
  "main.ts",
  "typings.d.ts"
]
```

I put them together.
```json
"files": [
  "main.ts",
  "typings.d.ts"
],
"filesGlob": [
  "main.ts",
  "typings.d.ts",
  "app/**/*.ts"  
]
```
That works for me.

**Comment by Yugloocamai** [comment]
@zetalei I followed your above example, but my atom-typescript still chokes on 
`moduleId: module.id`
I have @typings installed via npm

**Comment by Yugloocamai** [comment]
I found that if I add a reference to my top-level .ts file it works
` /// <reference path="../../node_modules/@types/node/index.d.ts" />`

**Comment by SirLants** [comment]
I'm following up on this, thanks to @Yugloocamai I was finally able to resolve this issue after spending hours on it. Can someone please explain how this issue hasn't been solved using all the other methods described in http://stackoverflow.com/questions/36700693/typescript-error-in-angular2-code-cannot-find-name-module and others? I feel like I shouldn't need to include a direct reference to those typings in my index.html file. 
Long story short, I'm glad that I was able to finally resolve this, but is there still something I'm missing?

**Comment by AnimaMundi** [comment]
With a brand new project I created today, I had to remove `"types": []` from `src/tsconfig.app.json` to fix this.

**Comment by stephengardner** [comment]
Brand new project created today June 18th and @AnimaMundi's solution was still necessary for me

**Comment by yanbu0** [comment]
Spun up a new CLI project and had to remove "types":[] from both tsconfig.app.json and tsconfig.server.json after enabling hmr.

**Comment by stephengardner** [comment]
Update December 2018, this `tsconfig.app.json` works:
```
{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "outDir": "../out-tsc/app",
    "baseUrl": "./",
    "module": "es2015",
    "types": ["node"]
  },
  "exclude": [
    "test.ts",
    "**/*.spec.ts"
  ]
}
```
